frontend/src/common/components/FormDialog.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Form } from 'react-final-form';
+
+// Material UI
+import {
+	Button,
+	Dialog,
+	DialogContent,
+	DialogTitle,
+	Grid,
+} from '@material-ui/core';
+
+// Util
+import ValidationService from '../services/validation.service';
+
+function FormDialog({
+	children,
+	close,
+	maxWidth,
+	open,
+	onCancel,
+	onSubmit,
+	primaryButtonLabel,
+	secondaryButtonLabel,
+	title,
+	validationSchema
+}) {
+
+	const generateErrors = (values) => ValidationService(values, validationSchema).errors;
+
+	return (
+		<Dialog fullWidth maxWidth={maxWidth} open={open} onClose={close}>
+			<DialogTitle>{title}</DialogTitle>
+			<DialogContent>
+				<Form
+					onSubmit={onSubmit}
+					validate={generateErrors}
+					render={({ handleSubmit }) => (
+						<form onSubmit={handleSubmit} noValidate>
```

Comment:
If you're curious about the final form syntax you can check out the following links:

- https://www.youtube.com/watch?v=Z8W102KMA-E
- https://codesandbox.io/s/react-final-form-material-ui-example-tqv09?file=/src/index.jsx:550-807